src/main/java/org/altbeacon/beacon/notifier/RegionMonitorNotifier.java

```diff
@@ -0,0 +1,66 @@
+package org.altbeacon.beacon.notifier;
+
+import org.altbeacon.beacon.MonitorNotifier;
+import org.altbeacon.beacon.Region;
+
+import java.util.ArrayList;
+
+/**
+ * Created by maxence on 17/04/15.
+ */
+public abstract class RegionMonitorNotifier extends ArrayList<Region> implements MonitorNotifier
```

Comment:
One consequence of this implementation is that the Regions list could be changed after monitoring is started.  And I believe the contains() method is not thread safe.  This could cause a ConcurrentModificationException if this method is called while code in another thread modifies the Regions in the list.